pkg/controller/deployment/deployment_controller.go

```diff
@@ -743,20 +849,21 @@ func (dc *DeploymentController) getNewReplicaSet(deployment *extensions.Deployme
 			Template: newRSTemplate,
 		},
 	}
-	// Set new replica set's annotation
-	setNewReplicaSetAnnotations(deployment, &newRS, newRevision)
 	allRSs := append(oldRSs, &newRS)
 	newReplicasCount, err := deploymentutil.NewRSNewReplicas(deployment, allRSs, &newRS)
 	if err != nil {
 		return nil, err
 	}
 
 	newRS.Spec.Replicas = newReplicasCount
+	// Set new replica set's annotation
+	setNewReplicaSetAnnotations(deployment, &newRS, newRevision)
```

Comment:
I finally decided not to have this update with this PR because it has been a pain in the ass to propagate the update without refactoring a bunch of code (tests included). I will open a separate issue about it if you are ok with it.